src/Utilities/functions.jl

```diff
@@ -1659,34 +1659,6 @@ function Base.:*(
     return g * f
 end
 
-# !!! warning
-#     MathOptInterface includes these methods to support coefficient types which
-#     are not subtypes of `Number`. We shoud consider removing them in MOI v2.0.
-
-function Base.:*(
-    f::Union{
-        MOI.ScalarAffineFunction{T},
-        MOI.ScalarQuadraticFunction{T},
-        MOI.VectorAffineFunction{T},
-        MOI.VectorQuadraticFunction{T},
-    },
-    g::T,
-) where {T}
-    return operate_coefficients(Base.Fix2(*, g), f)
-end
-
-function Base.:*(
-    f::T,
-    g::Union{
-        MOI.ScalarAffineFunction{T},
-        MOI.ScalarQuadraticFunction{T},
-        MOI.VectorAffineFunction{T},
-        MOI.VectorQuadraticFunction{T},
-    },
-) where {T}
-    return g * f
-end
```

Comment:
This is breaking, but only in the sense that it was never documented or expected to work. having functions with non-number coefficients is just asking for trouble.

It's also weird because this is the only case where we'd expect things to work. You can't do `*(::NonNumber, ::VariableIndex)`